drive_to_gps/src/current_heading.cpp

```diff
@@ -0,0 +1,69 @@
+#include "calculate_degrees.h"
+#include <drive_to_gps/CurrentHeadingConfig.h>
+#include <drive_to_gps/heading.h>
+#include <dynamic_reconfigure/server.h>
+#include <math.h>
+#include <ros/ros.h>
+#include <sensor_msgs/NavSatFix.h>
+
+/*
+Current heading is calculated in degrees with 0 degrees at North and positive is
+clockwise
+*/
+
+ros::Publisher pub_curr_head;
+sensor_msgs::NavSatFixPtr prev_gps = NULL;
+int refresh_rate = 1; // initialize 1hz as the loop rate (can be changed)
+
+void reconfigure_callback(drive_to_gps::CurrentHeadingConfig &config,
+                          uint32_t level) {
+  ROS_INFO("Rate = %d", config.refresh_rate);
+  refresh_rate = config.refresh_rate;
+}
+
+// determine the current heading of the rover
+void determine_curr_heading(const sensor_msgs::NavSatFixConstPtr &curr_gps) {
```

Comment:
maybe this callback should only update some buffer of navsat msgs

Reply:
maybe not a buff but a queue of navsat msgs. the sub cb can just add the msg to the end of the queue. if you move to using timer cb's, it could be done such that the cb deletes from the front of the queue until it finds a gps msgs that's x seconds from the current time (x being the configurable value that matches the publish frequency). i think itd give us a slightly less jittery heading